File: swift/HtmlConverter/Sources/HtmlConverter.swift
import Foundation

class HtmlConverter {
    func convertToHtml(_ fileName: String) throws -> String {
        let lines = try String(contentsOfFile: fileName, encoding: .utf8)
            .components(separatedBy: .newlines)

        // bug: should be "<html>"
        var html = "<html<"
        for line in lines {
            html += "<p>"
            html += HttpUtility.htmlEncode(line.trimmingCharacters(in: .whitespaces))
            // bug: should be "</p>"
            html += "<br />"
        }
        html += "</html>"
        return html
    }
}

File: CSharp/HtmlConverter-Kata/HtmlConverter.cs
using System.IO;
using System.Linq;

namespace HtmlConverterKata;

public class HtmlConverter
{
    public string ConvertToHtml(string filename)
    {
        using (TextReader unicodeFileStream = File.OpenText(filename))
        {
            // bug: should be "<html>"
            var html = "<html<";
            var line = unicodeFileStream.ReadLine();
            while (line != null)
            {
                html += "<p>";
                line = line.Trim();
                html += HttpUtility.HtmlEncode(line);
                // bug: should be "</p>"
                html += "<br />";
                line = unicodeFileStream.ReadLine();
            }

            html += "</html>";
            return html;
        }
    }

    
}